src/FixedSizeArrays.jl

```diff
@@ -134,37 +148,46 @@ macro fixed_vector(name, parent)
             end
         end
 
-        Base.@pure StaticArrays.Size{S}(::Type{$(name){S}}) = Size(S)
+        Base.@pure StaticArrays.Size{S}(::Type{$(name){S, Any}}) = Size(S)
         Base.@pure StaticArrays.Size{S,T}(::Type{$(name){S, T}}) = Size(S)
 
-        Base.@propagate_inbounds function Base.getindex(v::$(name), i::Integer)
+        Base.@propagate_inbounds function Base.getindex{S, T}(v::$(name){S, T}, i::Int)
             v.data[i]
         end
         @inline Base.Tuple(v::$(name)) = v.data
         @inline Base.convert{S, T}(::Type{$(name){S, T}}, x::NTuple{S, T}) = $(name){S, T}(x)
-        @inline Base.convert{SV <: $(name)}(::Type{SV}, x::StaticVector) = SV(x)
+        @inline Base.convert(::Type{$(name)}, x::StaticVector) = SV(x)
         @inline function Base.convert{S, T}(::Type{$(name){S, T}}, x::Tuple)
             $(name){S, T}(convert(NTuple{S, T}, x))
         end
 
-        @generated function StaticArrays.similar_type{SV <: $(name), T,S}(::Type{SV}, ::Type{T}, s::Size{S})
+        @generated function StaticArrays.similar_type{SV <: $(name), T, S}(::Type{SV}, ::Type{T}, s::Size{S})
             if length(S) === 1
                 $(name){S[1], T}
             else
                 StaticArrays.default_similar_type(T,s(),Val{length(S)})
             end
         end
-
-
+        size_or(::Type{$(name)}, or) = or
         eltype_or(::Type{$(name)}, or) = or
-        eltype_or{T}(::Type{$(name){TypeVar(:S), T}}, or) = T
-        eltype_or{S}(::Type{$(name){S, TypeVar(:T)}}, or) = or
-        eltype_or{S, T}(::Type{$(name){S, T}}, or) = T
 
-        size_or(::Type{$(name)}, or) = or
-        size_or{T}(::Type{$(name){TypeVar(:S), T}}, or) = or
-        size_or{S}(::Type{$(name){S, TypeVar(:T)}}, or) = (S,)
-        size_or{S, T}(::Type{$(name){S, T}}, or) = (S,)
+        if VERSION < v"0.6.0-dev"
```

Comment:
StaticArrays as a whole is now 0.6 only.  Is there any use in having this version check?